### ISSUE TITLE
Tapping tags crashes app

### ISSUE THREAD

**Issue by andyforceno** [wallabag/android-app#580]
I downloaded the latest version of the app available on the play store, but it crashes when I tap on any tags. I also noticed there are duplicate and triplicates for some tags. All tags crash the app, not one in particular. Running Wallabag 2.0.Is there a newer version of the app I should  be using instead?

**Comment by di72nn** [comment]
Does the system suggests you to send a crash report? If so, can you send it?

You're talking about the "Tags" screen available from the main menu, not about "Manage tags" screen available from article menu, right?

Do you experience crashes outside of the list of tags?

> Is there a newer version of the app I should be using instead?

There's no newer version yet.

**Comment by andyforceno** [comment]
Yes, the tags screen from the main menu u. An app crash report has been sent. 

_Sent from my Samsung SAMSUNG-SGH-I337 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by di72nn** [comment]
Thank you.

Dev note:
At the moment of writing I wasn't expecting to receive multiple tags with the same label, so this part of the app crashes.
Stacktrace:
```
org.greenrobot.greendao.DaoException: Expected unique result, but count was 2
	at org.greenrobot.greendao.AbstractDao.loadUnique(AbstractDao.java:170)
	at org.greenrobot.greendao.AbstractDao.loadUniqueAndCloseCursor(AbstractDao.java:159)
	at org.greenrobot.greendao.InternalQueryDaoAccess.loadUniqueAndCloseCursor(InternalQueryDaoAccess.java:41)
	at org.greenrobot.greendao.query.Query.unique(Query.java:130)
	at org.greenrobot.greendao.query.QueryBuilder.unique(QueryBuilder.java:479)
	at fr.gaulupeau.apps.Poche.ui.ArticleListFragment.resetContent(ArticleListFragment.java:151)
	at fr.gaulupeau.apps.Poche.ui.RecyclerViewListFragment.checkList(RecyclerViewListFragment.java:158)
	at fr.gaulupeau.apps.Poche.ui.RecyclerViewListFragment.onResume(RecyclerViewListFragment.java:106)
	at android.support.v4.app.Fragment.performResume(Fragment.java:2238)
	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1346)
	at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1528)
	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1595)
	at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:758)
	at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2363)
	at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2149)
	at android.support.v4.app.FragmentManagerImpl.optimizeAndExecuteOps(FragmentManager.java:2103)
	at android.support.v4.app.FragmentManagerImpl.execSingleAction(FragmentManager.java:1984)
	at android.support.v4.app.BackStackRecord.commitNowAllowingStateLoss(BackStackRecord.java:626)
	at android.support.v4.app.FragmentPagerAdapter.finishUpdate(FragmentPagerAdapter.java:143)
	at android.support.v4.view.ViewPager.populate(ViewPager.java:1268)
	at android.support.v4.view.ViewPager.populate(ViewPager.java:1116)
	at android.support.v4.view.ViewPager.onMeasure(ViewPager.java:1642)
	at android.view.View.measure(View.java:18596)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5827)
	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1435)
	at android.widget.LinearLayout.measureVertical(LinearLayout.java:721)
	at android.widget.LinearLayout.onMeasure(LinearLayout.java:612)
	at android.view.View.measure(View.java:18596)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5827)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:430)
	at android.view.View.measure(View.java:18596)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5827)
	at android.support.design.widget.CoordinatorLayout.onMeasureChild(CoordinatorLayout.java:714)
	at android.support.design.widget.HeaderScrollingViewBehavior.onMeasureChild(HeaderScrollingViewBehavior.java:90)
	at android.support.design.widget.AppBarLayout$ScrollingViewBehavior.onMeasureChild(AppBarLayout.java:1391)
	at android.support.design.widget.CoordinatorLayout.onMeasure(CoordinatorLayout.java:784)
	at android.view.View.measure(View.java:18596)
	at android.support.v4.widget.DrawerLayout.onMeasure(DrawerLayout.java:1081)
	at android.view.View.measure(View.java:18596)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5827)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:430)
	at android.support.v7.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:139)
	at android.view.View.measure(View.java:18596)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5827)
	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1435)
	at android.widget.LinearLayout.measureVertical(LinearLayout.java:721)
	at android.widget.LinearLayout.onMeasure(LinearLayout.java:612)
	at android.view.View.measure(View.java:18596)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5827)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:430)
	at android.view.View.measure(View.java:18596)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5827)
	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1435)
	at android.widget.LinearLayout.measureVertical(LinearLayout.java:721)
	at android.widget.LinearLayout.onMeasure(LinearLayout.java:612)
	at android.view.View.measure(View.java:18596)
	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5827)
	at android.widget.FrameLayout.onMeasure(FrameLayout.java:430)
	at com.android.internal.policy.impl.PhoneWindow$DecorView.onMeasure(PhoneWindow.java:2897)
	at android.view.View.measure(View.java:18596)
	at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2248)
	at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1306)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1548)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1191)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6642)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:777)
	at android.view.Choreographer.doCallbacks(Choreographer.java:590)
	at android.view.Choreographer.doFrame(Choreographer.java:560)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:763)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:145)
	at android.app.ActivityThread.main(ActivityThread.java:5951)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
```

**Comment by Strubbl** [comment]
What is the sense of having two tags with the same label? 

Do you expect to have two menu entries, but with the same label? 

Or one menu entry with the label, but displaying articles of both tags

**Comment by andyforceno** [comment]
Perhaps I don't understand how tagging works in Wallabag. I set up a number of tagging rules, some of which overlap. For instance, "github.com" is tagged as "Programming" while a site like "stackoverflow.com" is tagged "Programming,Linux" because that is mainly what I use it for. Is this why I see the duplicate tags in the android app? I don't see that in Wallabag itself, however. I was under the impression that this is how tags work, allowing you to classify one thing in multiple ways.

**Comment by di72nn** [comment]
> What is the sense of having two tags with the same label?

That's what server gives us.

> Do you expect to have two menu entries, but with the same label?

For starters, it shouldn't crash. I think tags with the same label should be virtually merged in lists.

@andyforceno as far as I know you did nothing wrong. The Android app wasn't ready for some of the server implementation quirks.

**Comment by Strubbl** [comment]
@andyforceno Thank you for the explanation. I do use tagging the same way.

**Comment by Strubbl** [comment]
Should be fixed since #588 has been integrated.
